src/app/services/validate-form.service.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@angular/core';
+import { ViewConfigItemComponent } from '../components/core/configuration-manager/view-config-item/view-config-item.component';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ValidateFormService {
+
+  constructor() { }
+
+  checkViewConfigItemFormValidity(viewComponent: ViewConfigItemComponent) {
+    if (viewComponent !== undefined) {
+      const isValidPassword = viewComponent.passwordMatched;
+      const isValidJSON = viewComponent.isValidJson;
+      const isValidFileExtension = viewComponent.isValidExtension;
+      const isValidForm = viewComponent.form.valid;
+      return isValidPassword && isValidForm && isValidJSON && isValidFileExtension;
```

Comment:
by default these are true? what if any of such item does not exist on `viewComponent`?

Reply:
By default all are true. There are member variables of viewComponent and will always be available until removed but yes value can be changes true/false